dnsmasq_interface.c

```diff
@@ -881,14 +909,6 @@ int FTL_listsfile(char* filename, unsigned int index, FILE *f, int cache_size, s
 	}
 	clearSetupVarsArray(); // will free/invalidate IPv6addr
 
-	// If no IPv4 address was found but user wants us to server NXDOMAIN
-	// we have to mock an IP record (which won't do anything in the end)
-	if(!has_IPv4 && config.blockingmode == MODE_NX)
-	{
-		if(inet_pton(AF_INET, "127.0.0.1", &addr4) > 0)
-			has_IPv4 = true;
-	}
-
```

Comment:
Why is this section deleted?

Reply:
This was meant for the case when there was no `IPV4_ADDRESS` in `setupVars.conf` as we need an IP for `NXDOMAIN` blocking. However, when now specifying `NXDOMAIN` or `NULL` as option for `BLOCKINGMODE` we will always use the hard-coded IP `0.0.0.0` and hence don't need a fallback solution any more.